framework/src/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -541,7 +517,14 @@ public Boolean visitDeclared_Typevar(AnnotatedDeclaredType subtype, AnnotatedTyp
 
     @Override
     public Boolean visitDeclared_Union(AnnotatedDeclaredType subtype, AnnotatedUnionType supertype, VisitHistory visited) {
-        return visitUnionSupertype(subtype, supertype, visited);
+        Types types = checker.getTypeUtils();
+        for (AnnotatedDeclaredType supertypeAltern : supertype.getAlternatives()) {
+            if (TypesUtils.isErasedSubtype(types, subtype.getUnderlyingType(), supertypeAltern.getUnderlyingType())
+                    && isSubtype(subtype, supertypeAltern)) {
```

Comment:
This is one of the few places that uses an `&&` to combine the two subtype tests. Below most of them are separate `if`s. Is there a reason for this?